resolwe_bio/processes/clustering/pca.py

```diff
@@ -0,0 +1,286 @@
+        except ValueError:
+            error("Cannot apply Z-score normalization to expression values.")
+
+    return expressions
+
+
+def get_csv(fnames):
+    """Read CSV files, perform transformations and return Pandas DataFrame."""
+    expressions = [read_csv(fname) for fname in fnames]
+    return pd.concat(expressions, axis=1, join="inner")
+
+
+def run_pca(
+    exp_paths,
+    rc_paths,
+    sample_ids,
+    gene_labels,
+    log2_transform,
+    standard_scaler,
+    error,
+):
+    """Read data, run PCA, and output results."""
+    expressions = get_csv(exp_paths)
+    # Filter out genes with low expression values
+    if rc_paths:
+        read_counts = get_csv(rc_paths)
+        percentile_80 = np.percentile(read_counts, 80, axis=1)
+        retained_genes = percentile_80 > 100
+        expressions = expressions.loc[retained_genes]
+
+    if expressions.empty:
+        error(
+            "Gene selection and filtering resulted in no genes. Please select different samples or genes."
+        )
+
+    expressions = transform(
+        error=error,
+        expressions=expressions,
+        log2=log2_transform,
+        z_score=standard_scaler,
+    )
+
+    if gene_labels:
+        gene_labels = list(set(gene_labels).intersection(expressions.index))
+        gene_labels.sort()
+        expressions = expressions.loc[gene_labels]
+
+    result = get_pca(expressions, gene_labels, error)
+    save_pca(result, sample_ids)
+
+
+class PrinicipalComponentAnalysis(Process):
+    """Principal component analysis process (beta)."""
+
+    slug = "pca-beta"
+    name = "Principal component analysis (beta)"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "public.ecr.aws/genialis/resolwebio/common:4.1.1",
+            },
+        },
+        "resources": {
+            "cores": 1,
+            "memory": 4096,
+            "storage": 10,
+        },
+    }
+    data_name = "{{ alignment|name|default('?') }}"
+    version = "0.1.0"
+    process_type = "data:pca"
+    category = "Enrichment and Clustering"
+    scheduling_class = SchedulingClass.INTERACTIVE
+    persistence = Persistence.TEMP
+
+    description = "Principal component analysis (PCA)"
+
+    class Input:
+        """Input fields."""
+
+        exps = ListField(DataField(data_type="expression"), label="Expressions")
+        genes = ListField(
+            StringField(),
+            label="Gene subset",
+            required=False,
+            placeholder="new gene id, e.g. ENSG00000185982 (ENSEMBL database)",
+            description="To subset your analysis to only a subset of genes, specify them here. "
+            "Specify at least two genes or leave this field empty.",
+        )
+        low_expression_filter = BooleanField(
+            label="Low expression filter",
+            default=True,
+            description="Filter the input expression matrix to remove genes "
```

Comment:
Please also specify that raw counts need to be available for the expression (shouldn't work for cases where only tpms have been uploaded, for example).